js/components/generate/GenerateForm.js

```diff
@@ -0,0 +1,397 @@
+import React from "react"
+
+import { Graph, populateHybridRelatives } from "../graph/Graph"
+import Disclaimer from "../common/Disclaimer"
+import { ErrorMessage } from "../common/react_modal.js.jsx"
+
+export default class GenerateForm extends React.Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      courses: "",
+      taken: "",
+      departments: "CSC, MAT, STA",
+      maxDepth: 0,
+      location: ["utsg"],
+      includeRaws: false,
+      includeGrades: false,
+      fceCount: 0,
+      showWarning: false,
+      invalidCourses: [],
+    }
+
+    this.graph = React.createRef()
+  }
+
+  setFCECount = credits => {
+    this.setState({ fceCount: credits })
+  }
+
+  incrementFCECount = credits => {
+    this.setState({ fceCount: this.state.fceCount + credits })
+  }
+
+  handleInputChange = event => {
+    const target = event.target
+    let value
+    if (target.type === "checkbox") {
+      value = target.checked
+    } else if (target.type === "select-multiple") {
+      value = Array.from(target.selectedOptions, option => option.value)
+    } else if (target.type === "number") {
+      value = parseInt(target.value)
+    } else {
+      value = target.value
+    }
+    const name = target.name
+
+    this.setState({
+      [name]: value,
+    })
+  }
+
+  handleSubmit = event => {
+    event.preventDefault()
+
+    // NOTE: Due to the aynchronous nature of state update, this does not have any effect in the running function,
+    // but plays a role in the next render
+    this.setState({ courses: this.state.courses.replaceAll(" ", "") })
+
+    if (!this.state.courses.length) {
+      this.setState({ showWarning: true, invalidCourses: [] })
```

Comment:
include a `return` after this line, since the rest of the code shouldn't execute

Reply:
The problem is the following scenario: 
1. A user enters a good input, e.g. `CSC110Y1, MAT237Y1`
2. Presses generate, generating a nice graph
3. makes the input blank ('')
4. Presses generate button again

**With a return, the old graph will stay, but without `return` it will be *correctly* rerendered to blank.** 

I thought this was the behavior we were after, although I do understand it is an additional server request. Let me know what is preferred.

Reply:
Right, it's fine to keep the old graph displayed. No need to rerender.

Reply:
@mimischly7 I think you missed the point of my reply. You should add the `return` I asked you to add.

Reply:
apologies for this, just fixed it